hs_core/views/utils.py

```diff
@@ -781,8 +782,92 @@ def move_or_rename_file_or_folder(user, res_id, src_path, tgt_path, validate_mov
             raise ValidationError("File/folder move/rename is not allowed.")
 
     istorage.moveFile(src_full_path, tgt_full_path)
+    rename_irods_file_or_folder_in_django(resource, src_full_path, tgt_full_path)
+    hydroshare.utils.resource_modified(resource, user, overwrite_bag=False)
+
+
+# TODO: modify this to take short paths not including data/contents
+def rename_file_or_folder(user, res_id, src_path, tgt_path, validate_move_rename=True):
+    """
+    Move or rename a file or folder in hydroshareZone or any federated zone used for HydroShare
```

Comment:
Comment here is not accurate, ```Move or rename``` -> "Rename", also input parameter ```validate_move_rename``` should be changed to ```validate_rename``` to eliminate confusion.